src/main/java/gov/cabinetoffice/gap/adminbackend/services/ApplicationFormSectionService.java

```diff
@@ -118,4 +118,35 @@ public void updateSectionStatus(final Integer applicationId, final String sectio
         this.applicationFormRepository.save(applicationForm);
     }
 
+    public void updateSectionOrder(final Integer applicationId, final String sectionId, final Integer increment) {
+        AdminSession session = HelperUtils.getAdminSessionForAuthenticatedUser();
+
+        ApplicationFormEntity applicationForm = this.applicationFormRepository.findById(applicationId)
+                .orElseThrow(() -> new NotFoundException("Application with id " + applicationId + " does not exist"));
+
+        if (!session.getGrantAdminId().equals(applicationForm.getCreatedBy())) {
+            throw new AccessDeniedException("User " + session.getGrantAdminId()
+                    + " is unable to access the application form with id " + applicationId);
+        }
+
+        List<ApplicationFormSectionDTO> sections = applicationForm.getDefinition().getSections();
+        ApplicationFormSectionDTO section = applicationForm.getDefinition().getSectionById(sectionId);
+        int index = sections.indexOf(section);
+
+        final int BOTTOM_ENTRY = sections.size() - 1;
+        final int ESSENTIAL_AND_ELIGIBILITY = 2;
+
+        if (increment < 0 && index <= ESSENTIAL_AND_ELIGIBILITY)
+            throw new FieldViolationException("sectionId", "Section is already at the top");
+
+        if (increment > 0 && index >= BOTTOM_ENTRY)
+            throw new FieldViolationException("sectionId", "Section is already at the bottom");
+
+        sections.remove(index);
+        sections.add(index + increment, section);
```

Comment:
I'm finding this validation a bit confusing to read. We could simplify it by first calculating the new sections index by doing `final int NEW_SECTION_INDEX = index + increment`

Then the validation becomes really straightforward. e.g. `if (NEW_SECTION_INDEX > SECTION_LIST_SIZE)`